x-pack/legacy/plugins/siem/public/components/inspect/index.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonIcon, EuiToolTip } from '@elastic/eui';
+import React from 'react';
+import { connect } from 'react-redux';
+import { ActionCreator } from 'typescript-fsa';
+
+import { inputsModel, inputsSelectors, State } from '../../store';
+import { InputsModelId } from '../../store/inputs/constants';
+import { inputsActions } from '../../store/inputs';
+
+import { ModalInspectQuery } from './modal';
+import * as i18n from './translations';
+
+interface OwnProps {
+  queryId: string;
+  inspectIndex: number;
+  title: string | React.ReactElement | React.ReactNode;
+}
+
+interface InspectButtonReducer {
+  id: string;
+  isInspected: boolean;
+  loading: boolean;
+  inspect: inputsModel.InspectQuery | null;
+  selectedInspectIndex: number;
+}
+
+interface InspectButtonDispatch {
+  setIsInspected: ActionCreator<{
+    id: string;
+    inputId: InputsModelId;
+    isInspected: boolean;
+    selectedInspectIndex: number;
+  }>;
+}
+
+type InspectButtonProps = OwnProps & InspectButtonReducer & InspectButtonDispatch;
+
+export const InspectButtonComponent = ({
+  inspect,
+  isInspected,
+  loading,
+  inspectIndex = 0,
+  queryId = '',
+  selectedInspectIndex,
+  setIsInspected,
+  title = '',
+}: InspectButtonProps) => (
+  <>
+    <EuiToolTip
+      delay="regular"
+      position="top"
+      content={i18n.TOOLTIP_CONTENT}
+      data-test-subj="inspect-tooltip"
+    >
+      <EuiButtonIcon
```

Comment:
```
checkPropTypes.js?:20 Warning: Failed prop type: EuiButtonIcon requires aria-label or aria-labelledby to be specified because icon-only
    buttons are screen-reader-inaccessible without them.
    in EuiButtonIcon (created by InspectButtonComponent)
    in InspectButtonComponent (created by Connect(InspectButtonComponent))
    in Connect(InspectButtonComponent)
```